joblib/test/test_parallel.py

```diff
@@ -10,6 +10,10 @@
 import sys
 import io
 import os
+import subprocess
+from threading import Timer
+import re
```

Comment:
Any particular reason why? I have to say I don't tend to follow a precise guideline with imports, aside from going from stdlib to less core packages with a new line between each group, for example in nilearn, I would do stdlib, numpy, scipy, scikit-learn, nilearn.